.github/actions/analyse-obs/action.yaml

```diff
@@ -0,0 +1,60 @@
+name: Run PVS-Studio analysis
+inputs:
+  pvsUsername:
+    description: PVS Studio Licence Username
+    required: true
+  pvsKey:
+    description: PVS Studio Licence Key
+    required: true
+  target:
+    description: Build Target
+    required: true
+  config:
+    description: Build Configuration
+    required: true
+runs:
+  using: composite
+  steps:
+    - name: 'Setup PVS Studio'
+      shell: pwsh
+      run: |
+        Invoke-WebRequest -Uri https://cdn.pvs-studio.com/PVS-Studio_setup.exe -OutFile ..\PVS-Studio_setup.exe
+        Start-Process -FilePath ..\PVS-Studio_setup.exe -ArgumentList "/VERYSILENT /SUPPRESSMSGBOXES /NORESTART /COMPONENTS=Core" -Wait
+
+    - name: Activate PVS Studio
+      shell: pwsh
+      run: |
+        & "C:\Program Files (x86)\PVS-Studio\PVS-Studio_Cmd.exe" credentials -u ${{ inputs.pvsUsername }} -n ${{ inputs.pvsKey }}
+
+    - name: Run PVS Analysis
+      shell: pwsh
+      run: |
+        $pvsParams = @(
+            "--progress"
+            "--disableLicenseExpirationCheck"
+            "-p", "${{ inputs.target }}"
+            "-c", "${{ inputs.config }}"
+            "-t", "${{ github.workspace }}\build_x64\obs-studio.sln"
+            "-o", "${{ github.workspace }}\analysis.plog"
+            "-C", "${env:GITHUB_ACTION_PATH}\obs.pvsconfig"
+        )
+        & "C:\Program Files (x86)\PVS-Studio\PVS-Studio_Cmd.exe" @pvsParams
```

Comment:
Should use `Invoke-External` here (which includes exit code handling already).

Reply:
Since the exit code will be pretty much always non-zero (since 256 means that it found something, see the [documentation](https://pvs-studio.com/en/docs/manual/0035/)) that would always throw, so I'd have to wrap it in a `try...catch` block anyway.

The best thing to do here would probably be to write a little script that actually handles the return codes properly and only throws for those that are actual errors.